### PR TITLE
Documentation/instance mocking

### PR DESCRIPTION
**Pull request by robertbasic**
Hi!

I converted my blog post (http://robertbasic.com/blog/mocking-hard-dependencies-with-mockery) into a cookbook entry. This one is for #398

### CODE REVIEW

File: docs/cookbook/mocking_hard_dependencies.rst [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+            $result = $service->callExternalService($param);
+
+            $this->assertSame('Tested!', $result);
+        }
+    }
+
+If we run this test now, it should pass. Mockery does it's job and our ``App\Service`` will use the mocked external service instead of the real one.
+
+The problem whit this is when we want to, for example, test the ``App\Service\External`` itself, or if we use that class somewhere else in our tests.
```
Comment (padraic) [inline docs/cookbook/mocking_hard_dependencies.rst]: whit => with ;)

File: docs/cookbook/mocking_hard_dependencies.rst [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+            $service = new \App\Service();
+
+            $result = $service->callExternalService($param);
+
+            $this->assertSame('Tested!', $result);
+        }
+    }
+
+If we run this test now, it should pass. Mockery does it's job and our ``App\Service`` will use the mocked external service instead of the real one.
```
Comment (padraic) [inline docs/cookbook/mocking_hard_dependencies.rst]: it's => its

File: docs/cookbook/mocking_hard_dependencies.rst [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+The problem whit this is when we want to, for example, test the ``App\Service\External`` itself, or if we use that class somewhere else in our tests.
+
+When Mockery overloads a class, because of how PHP works with files, that overloaded class file must not be included otherwise Mockery will throw a "class already exists" exception. This is where autoloading kicks in and makes our job a lot easier.
+
+To make this possible, we'll tell PHPUnit to run the tests that have overloaded classes in separate processes and to not preserve global state. That way we'll avoid having the overloaded class included more than once. Of course this has it's downsides as these tests will run slower.
```
Comment (padraic) [inline docs/cookbook/mocking_hard_dependencies.rst]: it's => its

File: docs/cookbook/mocking_hard_dependencies.rst [hunk 4]
```diff
@@ -0,0 +1,93 @@
+.. index::
+    single: Cookbook; Mocking Hard Dependencies
+
+Mocking Hard Dependencies
```
Comment (padraic) [inline docs/cookbook/mocking_hard_dependencies.rst]: Perhaps "new Keyword" in brackets for readers who might not quite get what a hard dependency is right away? They are out there...

**Comment by padraic** [comment]
Added some very minor comments. Thanks @robertbasic :)

**Comment by davedevelopment** [comment]
Thanks @robertbasic!

**Pull request merged**